test/happy/oauth2_capture_redirect_test.clj

```diff
@@ -5,9 +5,25 @@
             [happy.oauth2-credentials :as credentials]))
 
 (deftest wait-for-redirect-test
-  (-> (fn []
-        (Thread/sleep 100)
-        (http/get "http://localhost/redirect?code=CODE"))
-      (Thread. "redirect in a little while")
-      (.start))
-  (is (= "CODE" (r/wait-for-redirect {}))))
+  (is (= "CODE" (let [redirect-uri "http://localhost"
+                      simulate-redirect (fn [uri] (http/get (str uri "?code=CODE")))
+                      code (r/wait-for-redirect simulate-redirect {:redirect_uri redirect-uri})]
```

Comment:
This test becomes simpler and less brittle with the new callback function parameter (in general, tests with a sleep smell).